erpnext/stock/stock_ledger.py

```diff
@@ -781,13 +781,19 @@ def update_rate_on_purchase_receipt(self, sle, outgoing_rate):
 				d.db_update()
 
 	def update_rate_on_subcontracting_receipt(self, sle, outgoing_rate):
-		if frappe.db.exists(sle.voucher_type + " Item", sle.voucher_detail_no):
-			frappe.db.set_value(sle.voucher_type + " Item", sle.voucher_detail_no, "rate", outgoing_rate)
+		if frappe.db.exists("Subcontracting Receipt Item", sle.voucher_detail_no):
+			frappe.db.set_value("Subcontracting Receipt Item", sle.voucher_detail_no, "rate", outgoing_rate)
 		else:
 			frappe.db.set_value(
 				"Subcontracting Receipt Supplied Item", sle.voucher_detail_no, "rate", outgoing_rate
 			)
 
+		scr = frappe.get_doc("Subcontracting Receipt", sle.voucher_no, for_update=True)
+		scr.set_missing_values()
```

Comment:
This method will take time to execute and reposting will be slow. Is there a way we can speed up the execution of this method?